src/marketplace/ScannersView.js

```diff
@@ -1,9 +1,116 @@
 //@flow strict
 import * as React from 'react';
-class ScannersView extends React.Component<{||}> {
-    render()
-    {    return (
-            <h1>This is the Scanners View</h1>
+import {getTankComponent, verifyComponent} from '../armory/GetInventoryInfo.js';
+import Cookies from 'universal-cookie';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+|};
+let itemsForSale = [];
+class ScannersView extends React.Component<Props, State> {
+    constructor() {
+		super();
+		this.state={
+			//This tank id is the one the user is currently working on
+			userId: '',
+		}
+		this.getMarketSales();
+		this.getUserId();
+	}
+
+	//This gets us the user's id 
+	getUserId = async ():Promise<void> => {
+		const cookies = new Cookies();
+		const token = cookies.get('token');
+		const response = await fetch('/api/user/getUser/', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+				'x-auth-token': token
+			},
+		});
+		const body = await response.text();
+		const jsonObjectOfUser = JSON.parse(body);
+		this.setState({userId:jsonObjectOfUser._id});
+    };
```

Comment:
alright, I'm going to let you self-review at this point, because if you can't find things that are very clearly wrong here, we have issues